zebra-chain/src/amount.rs

```diff
@@ -324,6 +324,15 @@ where
     }
 }
 
+impl<'amt, C> std::iter::Sum<&'amt Amount<C>> for Result<Amount<C>>
+where
+    C: Constraint + std::marker::Copy + 'amt,
+{
+    fn sum<I: Iterator<Item = &'amt Amount<C>>>(iter: I) -> Self {
+        iter.copied().sum()
```

Comment:
I just decided to call the underlying `Sum<Amount>`, so the implementations always match.